### PR TITLE
add xgrads

### PR DESCRIPTION
**Pull request by miniufo**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by miniufo** [comment]
@conda-forge/help-python, ready for review!

File: recipes/xgrads/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: 927c84c615123ee4ddc98a499da70f06341092e911846735c6f822d6970ef621
+
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
```
Comment (BastianZim) [inline recipes/xgrads/meta.yaml:17]: It seems like this package could probably be noarch. Can you please check?

https://conda-forge.org/docs/maintainer/knowledge_base.html#noarch-builds
Easier way to generate the package: https://github.com/conda/grayskull

**Review by BastianZim — CHANGES_REQUESTED** [review]

**Comment by miniufo** [comment]
Hi @BastianZim, I've changed to noarch.  Could you please see if this is OK now?  Thank you for your kind help.

File: recipes/xgrads/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python >=3.6
+    - xarray
+    - dask
+    - numpy
+    - pyproj
+    - numba
```
Comment (BastianZim) [inline recipes/xgrads/meta.yaml:27]: This one is not listed upstream, is it really needed?

**Review by BastianZim — CHANGES_REQUESTED** [review]

File: recipes/xgrads/meta.yaml (hunk 2 above)
Comment (miniufo) [reply recipes/xgrads/meta.yaml:27]: Hi, @BastianZim It is necessary to run the tests and Jupiter notebooks. But not so for the package itself.

**Comment by miniufo** [comment]
Hi @BastianZim, could you check this again please?  Now the `utils.py` requires the packages listed in the `meta.yml`.  I am not sure what do you mean by upstream.  Could you provide the link so that I can resolve the conflict?

**Comment by BastianZim** [comment]
Upstream is just a term for the package source itself. Upstream is where this package is coming from, and downstream is any software using this package.

I'll have a look now.

**Comment by BastianZim** [comment]
@conda-forge-admin, please lint

**Pull request merged**